src/transformers/models/longformer/modeling_tf_longformer.py

```diff
@@ -2171,16 +2171,14 @@ def call(
 
         # set global attention on question tokens
         if inputs["global_attention_mask"] is None and inputs["input_ids"] is not None:
-            if inputs["input_ids"] is None:
-                logger.warning(
-                    "It is not possible to automatically generate the `global_attention_mask`. Please make sure that it is correctly set."
-                )
-            elif (
-                tf.where(inputs["input_ids"] == self.config.sep_token_id).shape[0] != 3 * inputs["input_ids"].shape[0]
+            if (
+                shape_list(tf.where(inputs["input_ids"] == self.config.sep_token_id))[0]
+                != 3 * shape_list(inputs["input_ids"])[0]
             ):
                 logger.warning(
                     f"There should be exactly three separator tokens: {self.config.sep_token_id} in every sample for questions answering. You might also consider to set `global_attention_mask` manually in the forward function to avoid this. This is most likely an error."
                 )
+                inputs["global_attention_mask"] = tf.fill(shape_list(inputs["input_ids"]), value=1)
```

Comment:
this doesn't look correct to me. The "default" global_attention_mask is all `0`s so I think it should be:

```python
inputs["global_attention_mask"] = tf.fill(shape_list(inputs["input_ids"]), value=0)
```

Also we could improve the warning a bit by appending a sentence like. `Disabling global attention for this forward pass...`